futurex_openedx_extensions/helpers/export_data.py

```diff
@@ -53,18 +49,26 @@ def get_filtered_query_params(self) -> dict:
         query_params.pop('page', None)
         return query_params
 
+    def get_serialized_fx_permission_info(self) -> dict:
+        """get serialized fx_permission info for task"""
+        fx_permission_info = self.request.fx_permission_info
+        user = fx_permission_info.get('user')
+        fx_permission_info.update({'user': user.id})
+        return fx_permission_info
+
     def generate_csv_url_response(self) -> dict:
         """Return response with csv file url"""
         filtered_query_params = self.get_filtered_query_params()
         view_url = self.get_view_request_url(filtered_query_params)
+        fx_permission_info = self.get_serialized_fx_permission_info()
         view_data = {
-            'user_id': self.request.user.id,
             'query_params': filtered_query_params,
             'kwargs': self.kwargs,
             'path': self.request.path,
         }
-        csv_file = export_data_to_csv(view_url, view_data, self.request.fx_permission_info, self.filename)
-        return {'download_url': self.build_download_url(csv_file)}
+        fx_task = FxTask.objects.create(filename=self.filename)
+        export_data_to_csv_task.delay(fx_task.id, view_url, view_data, fx_permission_info, self.filename)
```

Comment:
the `filename` will not be the same because it's a task and there will be millisecond delay
```suggestion
        exported_file_name = self.filename
        fx_task = FxTask.objects.create(filename=exported_file_name)
        export_data_to_csv_task.delay(fx_task.id, view_url, view_data, fx_permission_info, exported_file_name)
```